components/ComparisonRenameModal.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <b-modal :ref="modalRef" title="Rename comparison" class="text-center" @hidden="destroyModal" hide-footer centered size="lg">
+    <div class="text-center" style="max-height: 80vh; overflow: auto">
+			<h4>Rename "{{ oldComparisonName }}" to:</h4>
+			<b-form-input v-model="newComparisonName" required debounce="500" class="mt-2 mb-2"/>
+			<b-button @click="renameComparison" variant="success" :disabled="newNameInput" class="mb-1">
+				Confirm
+			</b-button>
+    </div>
+  </b-modal>
+</template>
+
+<script lang="ts">
+import {Segments}			from '~/util/splits';
+import {Component, Prop, mixins}  from 'nuxt-property-decorator';
+import BaseModal            			from '~/components/BaseModal.vue';
+import {whithLoad} 								from '~/util/loading';
+import store 											from '~/util/store';
+
+@Component
+export default class ComparisonRenameModal extends mixins(BaseModal) {
+  modalRef: string = 'ComparisonRenameModal';
+
+	newComparisonName: string = '';
+
+	segments: Segments = store.state.splitFile!.Run?.Segments;
+
+	@Prop()
+	oldComparisonName!: string;
+
+	@Prop()
+	callback!: Function;
+
+	get newNameInput() {
+		return this.newComparisonName === '';
```

Comment:
To make sure this is a non-empty string you can do 
```suggestion
		return this.newComparisonName.trim() === '';
```
`trim` will remove spaces at the beginning and the end of the string, this way if people enter only spaces it will still be considered empty. Technically you don't need `=== ''` either because empty strings are "falsy" (i.e. equivalent to `false` when used as a boolean) but you can keep it for clarity (also talking about clarity, the name of this getter doesn't really suggest it's a boolean thing, it sounds like it returns a string or something, but that's a small detail)